qemu/tests/eject_media.py

```diff
@@ -24,90 +24,78 @@ def run(test, params, env):
-    if new_img_name not in str(blocks_info):
-        raise error.TestFail("Fail to chang cdrom to %s." % new_img_name)
-    if qmp_used:
-        eject_cmd = "eject device=%s, force=True" % device_name
-    else:
-        eject_cmd = "eject device=%s" % device_name
-    error.context("Eject device after add new image by change command")
-    vm.monitor.send_args_cmd(eject_cmd)
+    if not check_block(new_img_name):
+        test.fail("Fail to change cdrom to %s." % new_img_name)
+    error_context.context("Eject device after add new image by change command")
+    eject_cdrom(device_name)
     logging.info("Wait until new image is ejected")
     time.sleep(10)
+    if check_block(new_img_name):
+        test.fail("Fail to eject cdrom %s." % orig_img_name)
 
-    blocks_info = vm.monitor.info("block")
-    if new_img_name in str(blocks_info):
-        raise error.TestFail("Fail to eject cdrom %s." % orig_img_name)
-
-    error.context("Insert %s to device %s" % (orig_img_name, device_name))
-    change_cmd = "change device=%s,target=%s" % (device_name, orig_img_name)
-    vm.monitor.send_args_cmd(change_cmd)
+    error_context.context("Insert %s to device %s" % (orig_img_name, device_name))
+    change_media(device_name, orig_img_name)
     logging.info("Wait until device changed")
     time.sleep(10)
-    blocks_info = vm.monitor.info("block")
-    if orig_img_name not in str(blocks_info):
-        raise error.TestFail("Fail to change cdrom to %s." % orig_img_name)
+    if not check_block(orig_img_name):
+        test.fail("Fail to change cdrom to %s." % orig_img_name)
 
-    error.context("Try to eject non-removable device")
+    error_context.context("Try to eject non-removable device")
     p_dict = {"removable": False}
     device_name = vm.get_block(p_dict)
     if device_name is None:
-        raise error.TestFail("Could not find non-removable device")
-    if params.get("force_eject", "no") == "yes":
-        if not qmp_used:
-            eject_cmd = "eject -f %s " % device_name
-        else:
-            eject_cmd = "eject device=%s, force=True" % device_name
-    else:
-        eject_cmd = "eject device=%s," % device_name
+        test.fail("Could not find non-removable device")
     try:
-        vm.monitor.send_args_cmd(eject_cmd)
+        if params.get("force_eject", "no") == "yes":
+            eject_cdrom(device_name, force=True)
+        else:
+            eject_cdrom(device_name)
     except Exception, e:
         if "is not removable" not in str(e):
-            raise error.TestFail(e)
+            test.fail(e)
         logging.debug("Catch exception message: %s" % e)
     logging.info("Wait until device is ejected")
     time.sleep(10)
-    blocks_info = vm.monitor.info("block")
-    if device_name not in str(blocks_info):
-        raise error.TestFail("Could remove non-removable device!")
+    if not check_block(device_name):
+        raise test.fail("Could remove non-removable device!")
```

Comment:
<del>`raise`</del>
Please provide more detailed commit message elaborate some your changes.

Reply:
please remove `raise`

Reply:
Yes, please remove `raise` since it is a function not an error.